### PR TITLE
EventGrid Event raise on binary type

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/13581

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py [hunk 1]
```diff
@@ -88,3 +89,15 @@ def _is_cloud_event(event):
         return all([_ in event for _ in required]) and event['specversion'] == "1.0"
     except TypeError:
         return False
+
+def _eventgrid_event_options(event):
```
[the remark this answers is not in the document]
Comment (lmazuel) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py:93]: +1 for @KieranBrantnerMagee 's comment

[the remark this answers is not in the document]
Comment (lmazuel) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py:93]: `_eventgrid_data_typecheck`?

**Review by KieranBrantnerMagee — APPROVED** [review]
lgtm

**Pull request merged**